### DISCUSSION TITLE
Question about testing plugin.ts

### DISCUSSION THREAD

**Discussion by henry-wu-1130** [aarongarciah/figma-plugin-typescript-boilerplate#41] [Q&A]
really thanks for this boilerplate.

but i also have some questions here that i'm curious about how to test function in plugin.ts file?

i have trying to use jest in my project but when running, it showing that " ReferenceError: figma is not defined".

i found that in this project. you only test for build script. what's the reason? and if i want to test plugin.ts, should i mock figma object and it's corresponding properties. or is there any better way to do this?

thanks, and have a nice day

**Comment by aarongarciah** [comment]
@henry-wu-1130

The boilerplate didn't focus on testing the implementation of the plugin, just putting some testing setup in place. But as said, there's no way to test the real implementation of the plugin since it tuns inside a Figma worker (which injects the `figma` object at runtime).

TL;DR: yes, you need to mock the `figma` object if you want to test what's inside `plugin.ts`. I haven't though about how proper testing would look like. Maybe some existing plugins have some real testing to take inspiration from.

(Sorry for the delay, I totally missed this notification)
